commands/pipelines/setup.js

```diff
@@ -159,6 +159,65 @@ function* getCISettings (yes, organization) {
   return settings
 }
 
+function createApps (heroku, archiveURL, pipeline, pipelineName, stagingAppName, organization) {
+  const prodAppSetupPromise = createApp(heroku, {
+    archiveURL,
+    pipeline,
+    name: pipelineName,
+    stage: 'production',
+    organization
+  })
+
+  const stagingAppSetupPromise = createApp(heroku, {
+    archiveURL,
+    pipeline,
+    name: stagingAppName,
+    stage: 'staging',
+    organization
+  })
+
+  const promises = [prodAppSetupPromise, stagingAppSetupPromise]
+
+  return Promise.all(promises).then(appSetups => {
+    return appSetups
+  }, (error) => {
+    cli.exit(1, error)
+  })
+}
+
+function pollAppStatus (heroku, build) {
+  return new Promise((resolve, reject) => {
+    const count = setInterval(() => {
```

Comment:
I'd rather see this implemented with a recursive `setTimeout` than a `setInterval`. Currently we could kick off another round of the poll loop before the previous call to `api.getAppSetup` had completed.